runtime/compiler/optimizer/DataAccessAccelerator.cpp

```diff
@@ -196,6 +196,20 @@ int32_t TR_DataAccessAccelerator::performOnBlock(TR::Block* block, TreeTopContai
    int32_t blockResult = 0;
    bool requestOSRGuardRemoval = false;
 
+   /* Disable DAA if vector packed decimal facilities aren't available.
+    * Disabling the DAA optimization for non-vectorized paths temporarily to unblock
+    * off-heap enablement while we continue to investigate the bug in non-vectorized
+    * pdload evaluator.
+    */
+   if (comp()->target().cpu.isZ() && TR::Compiler->om.isOffHeapAllocationEnabled()
+      && !(comp()->target().cpu.supportsFeature(OMR_FEATURE_S390_VECTOR_PACKED_DECIMAL)
```

Comment:
Is it possible that we could not have `OMR_FEATURE_S390_VECTOR_PACKED_DECIMAL` support but could have `OMR_FEATURE_S390_VECTOR_PACKED_DECIMAL_ENHANCEMENT_FACILITY` or other extension which was introduce in later hardware ?

Reply:
Not sure about that, I don't see anything about that in the PoPs. But `pdload` needs `OMR_FEATURE_S390_VECTOR_PACKED_DECIMAL` so if that isn't there but other facilities are, we would be forced to go down the non-vectorized path for `pdload`.

Reply:
Quoting the PoPs chapter 1-29 here [1]
> The vector-packed-decimal-enhancement facility 1
may be available on models implementing the
z/Architecture. The facility extends the vector
packed-decimal facility. 

Same goes for facility 2 and 3. 

[1]. https://www.ibm.com/docs/en/module_1678991624569/pdf/SA22-7832-13.pdf?cp=SSFC5Y

Reply:
ah, yup, saw that bit but wasn't sure if extends also implied base facility must be there 😅